src/collection.js

```diff
@@ -230,6 +237,52 @@ export default class Collection {
   }
 
   /**
+   * Validate the passed hook is correct.
+   *
+   * @param {Array|undefined} hook.
+   * @return {Array}
+   **/
+  _validateHook(hook) {
+    if (!Array.isArray(hook)) {
+      throw new Error("A hook definition should be an array of functions.");
+    }
+    return hook.map(fun => {
+      if (typeof fun !== "function") {
+        throw new Error("A hook definition should be an array of functions.");
+      }
+      return fun;
+    });
+  }
+
+  /**
+   * Validates a list of hooks.
+   *
+   * @param  {Object|undefined} hooks
+   * @return {Object}
+   */
+  _validateHooks(hooks) {
+    if (typeof hooks === "undefined") {
+      return {};
+    }
+    if (Array.isArray(hooks)) {
+      throw new Error("hooks should be an object, not an array.");
```

Comment:
I've updated the code to act like you say, but it means that when registering the collection hooks I need to now register a "factory" to access the collection. Something like this:

```js
function signatureVerifierFactory(collection) {
  return function validateCollectionSignature(payload) {
    // Access collection here.
  };
};
```

That's doable, but I wonder if it's the API we want to expose?